dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAIChatCompletion_AutoFunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,379 @@
+
+        // Assert
+        Assert.NotNull(result);
+
+        Assert.Empty(invokedFunctions);
+
+        var functionCalls = FunctionCallContent.GetFunctionCalls(result);
+        Assert.NotNull(functionCalls);
+        Assert.Single(functionCalls);
+
+        var functionCall = functionCalls.First();
+        Assert.Equal("DateTimeUtils", functionCall.PluginName);
+        Assert.Equal("GetCurrentDate", functionCall.FunctionName);
+    }
+
+    [Fact]
+    public async Task SpecifiedInCodeInstructsConnectorToInvokeKernelFunctionAutomaticallyForStreamingAsync()
+    {
+        // Arrange
+        this._kernel.ImportPluginFromType<DateTimeUtils>();
+
+        var invokedFunctions = new List<string>();
+
+        this._autoFunctionInvocationFilter.RegisterFunctionInvocationHandler(async (context, next) =>
+        {
+            invokedFunctions.Add(context.Function.Name);
+            await next(context);
+        });
+
+        var settings = new OpenAIPromptExecutionSettings { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto(autoInvoke: true) };
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("How many days until Christmas?");
+
+        string result = "";
+
+        // Act
+        await foreach (var content in this._chatCompletionService.GetStreamingChatMessageContentsAsync(chatHistory, settings, this._kernel))
+        {
+            result += content;
+        }
+
+        // Assert
+        Assert.NotNull(result);
+
+        Assert.Single(invokedFunctions);
+        Assert.Contains("GetCurrentDate", invokedFunctions);
+    }
+
+    //[Fact]
+    //public async Task SpecifiedInPromptInstructsConnectorToInvokeKernelFunctionAutomaticallyForStreamingAsync()
+    //{
+    //    // Arrange
+    //    this._kernel.ImportPluginFromType<DateTimeUtils>();
+
+    //    var invokedFunctions = new List<string>();
+
+    //    this._autoFunctionInvocationFilter.RegisterFunctionInvocationHandler(async (context, next) =>
+    //    {
+    //        invokedFunctions.Add(context.Function.Name);
+    //        await next(context);
+    //    });
+
+    //    var promptTemplate = """"
+    //        template_format: semantic-kernel
+    //        template: How many days until Christmas?
+    //        execution_settings:
+    //          default:
+    //            temperature: 0.1
+    //            function_choice_behavior:
+    //              type: auto
+    //        """";
+
+    //    var promptFunction = KernelFunctionYaml.FromPromptYaml(promptTemplate);
+
+    //    string result = "";
+
+    //    // Act
+    //    await foreach (string c in promptFunction.InvokeStreamingAsync<string>(this._kernel))
+    //    {
+    //        result += c;
+    //    }
+
+    //    // Assert
+    //    Assert.NotNull(result);
+
+    //    Assert.Single(invokedFunctions);
+    //    Assert.Contains("GetCurrentDate", invokedFunctions);
+    //}
```

Comment:
Commented ? Should remove or skip?

Reply:
They will be needed later to test other parts of the new function calling model when they are added one after another. It'll be challenging to keep track of tests that were copied from the old big function calling abstraction PR and which were not. So, I'll keep them commented out for a while.